core/src/main/java/fr/sncf/osrd/new_infra_state/api/NewTrainPath.java

```diff
@@ -0,0 +1,44 @@
+package fr.sncf.osrd.new_infra_state.api;
```

Comment:
It looks like this file should be in `new_infra_state.implementation`.

Reply:
We already have a few records in `api` modules for classes that are just collections of fields.

The utility function makes it a bit different, but it's just a convenient way to handle an `ImmutableList<LocatedElement>`, it's not directly related to an instance of `NewTrainPath`. I don't think it would make sense in an interface.